arrows/vxl/tests/test_image_io.cxx

```diff
@@ -0,0 +1,95 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/**
+ * \file
+ * \brief test VXL image class functionality
+ */
+
+#include <test_gtest.h>
+#include <test_tmpfn.h>
+
+#include <arrows/tests/test_image.h>
+
+#include <arrows/vxl/image_container.h>
+#include <arrows/vxl/image_io.h>
+
+#include <kwiversys/SystemTools.hxx>
+#include <vital/plugin_loader/plugin_manager.h>
+#include <vital/util/transform_image.h>
+
+#include <gtest/gtest.h>
+
+namespace kv = kwiver::vital;
+namespace ka = kwiver::arrows;
+using ST = kwiversys::SystemTools;
+
+kv::path_t g_data_dir;
+static std::string test_color_image_name = "images/small_color_logo.png";
+static std::string test_plane_image_name = "images/planes_logo.png";
+
+// ----------------------------------------------------------------------------
+int
+main( int argc, char** argv )
+{
+  ::testing::InitGoogleTest( &argc, argv );
+  TEST_LOAD_PLUGINS();
+
+  GET_ARG( 1, g_data_dir );
+
+  return RUN_ALL_TESTS();
+}
+
+// ----------------------------------------------------------------------------
+class image_io : public ::testing::Test
+{
+  TEST_ARG( data_dir );
+};
+
+// ----------------------------------------------------------------------------
+TEST_F(image_io, save_plane)
+{
+  // Create an image to output
+  auto const& vil_image = vil_image_view< vxl_byte >( 150, 150, 3 );
+  auto image = ka::vxl::image_container( vil_image );
+  auto image_ptr = std::make_shared< ka::vxl::image_container >( image );
+
+  // Configure to split channels
+  ka::vxl::image_io writer;
+  auto config = kv::config_block::empty_config();
+  config->set_value( "split_channels", true );
+  writer.set_configuration( config );
+
+  auto const& output_filename =
+    kwiver::testing::temp_file_name( "image_io_save_plane-", ".png" );
+
+  std::cout << output_filename << std::endl;
+
+  writer.save( output_filename, image_ptr );
+
+  const auto saved_filenames = writer.plane_filenames( output_filename );
+
+  for( auto const& saved_filename : saved_filenames )
```

Comment:
It seems the only thing this tests is that the code executed without crashing. Is there any way to verify it actually did what it is supposed to do?

Reply:
I updated this to perform a save->reload->compare operation. I'm not thrilled about pulling in the `load` functionality when trying to test `save`, but given there's a load-specific test, this seems acceptable.